cmd/helm/package.go

```diff
@@ -122,8 +123,8 @@ func (p *packageCmd) run(cmd *cobra.Command, args []string) error {
 		debug("Setting version to %s", p.version)
 	}
 
-	if filepath.Base(path) != ch.Metadata.Name {
-		return fmt.Errorf("directory name (%s) and Chart.yaml name (%s) must match", filepath.Base(path), ch.Metadata.Name)
```

Comment:
^ We should probably leave that check in because filepath.Base(path) will also remove any trailing path separators at the end of the path before returning the last element.